raycaster_float.cpp

```diff
@@ -1,7 +1,41 @@
 // floating-point implementation for testing/comparison
+//
+// This diagram show the coordinate system used in the
+// RayCasterFloat::Distance() function.
+//
+//                        W
+//                        xO   rayA    Wd
+//                        0X          WON
+//   sin-                 0X         NO      sin+
+//   cos+             WXXXk0XXXW    XK       cos+
+//   tan-         XXXXXW  0X  WXXXKkX        tan+
+//             NKKW       0X     WONKKN
+//           W0X          0X    X0    X0W
+//          X0            0X   KX       0N
+//         XK             0X  ON         0N
+//         O              0XNOW           O
+//        XK              0O0             ON
+// WXXXXXXXx0XXXXXXXXXXXXXXkxXXXXXXXXXXXXXXOOXXXXXOO
+//        XK              0X              OW
+//         O              0X              O
+//         N0             0X             OW
+//          N0            0X            ON
+//           W0X          0X          K0
+//   sin-      WKKN       0X       NKKW      sin+
+//   cos-         NKXXXN  0X WNXXXKN         cos-
+//   tan+              NXXk0XXW              tan-
+//                        0X
+//                        0X
+//                        0X
+//                        W
+
 
 #include "raycaster_float.h"
 #include <math.h>
+#include <algorithm>
+
+#define P2P_DISTANCE(x1, y1, x2, y2) \
+    sqrt((float) ((x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2)))
```

Comment:
Inside the macro, you should always use `(x1)` rather than `x1`. Please be aware of the expansion.